.infrastructure/helm-chart/todoapp/templates/hpa.yml

```diff
@@ -0,0 +1,25 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{.Chart.Name}}
+  minReplicas: {{.Values.todoapp.hpa.minReplicas}}
+  maxReplicas: {{.Values.todoapp.hpa.maxReplicas}}
```

Comment:
Ensure that `{{.Values.todoapp.hpa.minReplicas}}` and `{{.Values.todoapp.hpa.maxReplicas}}` are defined in your `values.yaml` file. If they're not defined, it could lead to autoscaling issues.